plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -380,6 +408,10 @@ public static ColumnMapping oracleTimestampColumnMapping()
                 TIMESTAMP_MILLIS,
                 (resultSet, columnIndex) -> {
                     LocalDateTime timestamp = resultSet.getObject(columnIndex, LocalDateTime.class);
+                    // Adjust years when the value is B.C. dates because Oracle returns +1 year unless converting to string in their server side
+                    if (timestamp.getYear() <= 0) {
```

Comment:
Add a test with year = 0. Current tests are either negative years (-0001) or positive (0001).

Reply:
Year 0 case already exists in https://github.com/trinodb/trino/pull/10380/files#diff-b68241f0665724082f629407086d5eb4595bbe62206cb526fa9ad904b235dcc7R726-R728